bumble/core.py

```diff
@@ -90,7 +90,7 @@ class BaseError(BaseBumbleError):
 
     def __init__(
         self,
-        error_code: Optional[int],
+        error_code: int | None,
```

Comment:
Using | for unions is only supported for Python >= 3.10, so we can't use it yet.